exercises/collatz-conjecture/example.exs

```diff
@@ -0,0 +1,25 @@
+defmodule CollatzConjecture do
+
+  @doc """
+  calc/1 takes number (> 1), and returns the number of steps required to get to 1 when
+  following the rules: if n is odd, multiply with 3 and add 1. if n is even, divide by 2
+  """
+  @spec calc(number :: integer) :: pos_integer
+  def calc(number) when number < 1 do
+    raise ArgumentError, "Only positive numbers are allowed"
+  end
+
+  def calc(number) do
```

Comment:
Personally, if I were using a library and it provided a type signature for a function and I don't conform to that documentation, then I should expect a MatchError. That seems like reasonable behavior to me. Then we could also be more clear in the typespec for `calc/1` by saying `@spec calc(number :: pos_integer) :: pos_integer`, which I think would be a good thing.

I actually think it would be a _really_ great thing to show folks how to explicitly allow for a MatchError by changing the test to say `assert_raise MatchError, fn -> CollatzConjecture.calc(-15) end`. I don't think we have any other exercises about that yet, and it would be a good way to help folks explore pattern matching errors and typespecs.